V1 (#1)
multipath-tcp/mptcp-check: README.md

# mptcp-check
A website to check the validity of a mptcp connection

The website is build as a Flask app behind a lighttpd server.
Everything is package in a docker container.

## installing

> you will need to create the too following scripts. I would recommend the following. Do not forget to replace the `email` and `url`

Reviewer: ```suggestion
You will need to create the two following scripts. I would recommend the following. Do not forget to set `email` and `url`.
```

`start.sh`:
``` bash
#!/bin/bash
email= #the email used by certbot
url= #your domain name
path=/var/docker/mptcp-check

docker run --rm --name certbot-init \

>             -v "$path/cert:/etc/letsencrypt:rw" \

Reviewer: in bash, it is often recommended to use `{}` around variable names (`${path}`) to avoid typos (`$path-ok` is using the var `path` and add `-ok`, but if `_` is used, that's different), and clearly delimit the variables names.

            -p 80:80 \
            certbot/certbot certonly \
                --non-interactive --agree-tos \
                --email $email \
                -d $url \
                --standalone

docker run --name mptcp-check \
            -v "$path/www:/var/www/:ro" \
            -v "$path/cert/:/etc/letsencrypt/:ro" \

>             -p 80:80 -p 443:443 \

Reviewer: did you check the port 80 was OK?

Author: it seems to only not work with curl, I have no clue why.

Reviewer: I know that with a `.dev`, the upper layer forces to use `https`: https://ma.ttias.be/chrome-force-dev-domains-https-via-preloaded-hsts/

So maybe the browsers don't even try to use the port 80?

Here, we have an ICMP error, it looks like there is an issue with the firewall:

```
15:54:15.625189 IP 192.168.178.52.52850 > 5.196.67.207.80: Flags [S], seq 4252628027, win 32120, options [mss 1460,sackOK,TS val 1612797082 ecr 0,nop,wscale 7], length 0
15:54:15.665430 IP 5.196.67.207 > 192.168.178.52: ICMP 5.196.67.207 tcp port 80 unreachable, length 68
```

Author: since it is not related to the configuration, I will close the conversation.

            --restart unless-stopped \
            --network host \
            --detach \

>             mux514/mptcp-check:main

Reviewer: ```suggestion
            mptcp/mptcp-check:main
```

```

------------------------------------
`renew.sh`
``` bash
#!/bin/bash
email= #the email used by certbot
url= #your domain name
path=/var/docker/mptcp-check

> cert-before=$(sha256sum certbot/conf/live/$url/fullchain.pem)

Reviewer: ```suggestion
cert="certbot/conf/live/${url}/fullchain.pem"
cert_before=$(sha256sum "${cert}")
```

Don't use `-` in var names in bash, and use a var when a string is used twice to avoid typos

docker run --name certbot-renew \
    -v "$path/www:/var/www/:rw" \
    -v "$path/cert/:/etc/letsencrypt/:rw" \
    --detach \
    certbot/certbot \
    certonly --non-interactive --agree-tos --webroot \
            --webroot-path /var/www/ \
            --email $email \
            -d $url

> cert-after=$(sha256sum certbot/conf/live/$url/fullchain.pem)

Reviewer: ```suggestion
cert_after=$(sha256sum "${cert}")
```

Don't use `-` in var names in bash.

> if [ cert-before != cert-after ] then

Reviewer: ```suggestion
if [ "${cert_before}" != "${cert_after}" ]; then
```

Maybe better to use variables instead of strings? And `;` was missing.

    docker restart mptcp-check
fi
```
When the second one has been created, is needs to be added to the crontab and
run frequently.

To create the crontab:
```
crontab -e
```

> write `0 0 * * * root /path/to/renew.sh` on a new line

Reviewer: Don't use midnight: many people are doing that: more chance to have an error if there are too many requests.

```suggestion
write `23 2 * * * root /path/to/renew.sh` on a new line
```

## static files
The server will serve all files and folders in the `/var/www/files` directory.
You can use the following command to create dummy random files of various sizes.

> ```
> mkdir -p /var/docker/mptcp-check/www/files
> for i in 1M 10M 100M 1000M; do
>     head -c "${i}" /dev/urandom > "/var/docker/mptcp-check/www/files/${i}"
> done
> ```

Reviewer: ```suggestion
```bash
mkdir -p /var/docker/mptcp-check/www/files && cd $_
for i in 1M 10M 100M 1000M; do
    head -c "${i}" /dev/urandom > "${i}"
done
```
```

## updating
run the following commands
```

> docker pull mux514/mptcp-check:main

Reviewer: ```suggestion
docker pull mptcp/mptcp-check:main
```

docker stop mptcp-check
docker rm mptcp-check
```
then run the `start.sh` script again.

## update needs

> Currently, the apt version of lighttpd is 1.4.63, when the 1.4.76 is available, the last line of the `Dockerfile` should be swapped with the comment above.

Reviewer: ```suggestion
Currently, the apt version of lighttpd is 1.4.63. When the 1.4.76 or newer version is available, `mptcpize run` can be removed from the `Dockerfile` file, and the option `"server.network-mptcp"` can be set in the config.
```
